nixos/modules/services/security/tang.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.tang;
+in
+{
+  options.services.tang = {
+    enable = mkEnableOption "tang";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tang;
+      defaultText = literalExpression "pkgs.tang";
+      description = mdDoc "The tang package to use.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 7654;
+      description = mdDoc "The port to listen on.";
+    };
+
+  };
+  config = mkIf cfg.enable {
+    users = {
+      users.tang = {
+        group = "tang";
+        isSystemUser = true;
+      };
+      groups.tang = { };
+    };
```

Comment:
Is it not possible to use DynamicUser instead?

Reply:
Yep good call, done